### ISSUE TITLE
UselessConstantTypeHint should allow to specify array types

### ISSUE THREAD

**Issue by davidkmenta** [slevomat/coding-standard#978]
As a constant may contain an array, an annotation sometimes can be useful. Consider this:

```php
/**
 * @var array<string, int>
 */
public const LIST_OF_IDS = [
    'Some description of the ID' => 1,
    // ...
];

/**
 * @var string[]
 */
public const MESSAGES = [
    'Awesome message',
    // ...
];
```

Therefore it would be great if the sniff `SlevomatCodingStandard.TypeHints.UselessConstantTypeHint` would be configurable 😊.

**Comment by kukulich** [comment]
I still think the annotation is useless. It's constant so the types are clear and could not be changed.

**Comment by davidkmenta** [comment]
Ok :)
